### PR TITLE
feat(cli): restructure CLI into new hierarchy

### PR DESCRIPTION
**Pull request by bstrie**
Implements the CLI restructuring proposed in https://github.com/enarx/enarx/issues/1953 . This PR doesn't add any new functionality, it only moves existing commands to new places in the CLI and adds stub implementations of the new commands that have yet to be implemented (these stub implementations are hidden from the help output so that users do not accidentally attempt to use them; commands will be individually un-hidden as they are implemented). This also makes individual command handling more regular by moving functionality out of main.rs and into dedicated files for each subcommand, which is important for keeping things manageable since we're planning to add a lot of new commands.

To reiterate, don't be dismayed by the size of this PR, as most of the code here is code that already exists and is just being moved into new locations for better organization (and the rest of the new code is largely provisional for now), so it probably only needs to be lightly reviewed.

(Note also that this does not merge `enarx run` and `enarx deploy` as proposed in #1953; I figure that's a large enough change that it's better left for later.)

Signed-off-by: bstrie <865233+bstrie@users.noreply.github.com>

### CODE REVIEW

File: src/backend/sev/mod.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ use sallyport::item::{self, Item};
 mod builder;
 mod cpuid_page;
 mod data;
-mod snp;
+pub mod snp;
```
Comment (rvolosatovs) [inline src/backend/sev/mod.rs:27]: See https://enarx.dev/docs/Contributing/Coding-Style#use-declarations

Comment (rvolosatovs) [inline src/backend/sev/mod.rs:27]: see https://enarx.dev/docs/Contributing/Coding-Style#use-declarations

**Comment by rvolosatovs** [comment]
I will do a proper review once this is done!

**Comment by bstrie** [comment]
From this point forward I'll push any changes as new commits, to make it easier to review. I'll squash those once this is ready for approval.

File: src/backend/sev/mod.rs (hunk 1 above)
Comment (bstrie) [reply src/backend/sev/mod.rs]: Done.

**Review by rvolosatovs — APPROVED** [review]
Went through the PR on a call

**Pull request merged**